mesonbuild/scripts/env2mfile.py

```diff
@@ -160,7 +160,7 @@ def deb_compiler_lookup(infos: MachineInfo, compilerstems: T.List[T.Tuple[str, s
             pass
 
 def detect_cross_debianlike(options: T.Any) -> MachineInfo:
-    if options.debarch is None:
+    if options.debarch == 'auto':
```

Comment:
"add autodetect" -- more like fix the current broken one. :D Looks like we only evaluate this function if options.debarch is not None, and then the first thing we do is check if it is None. Which cannot ever be the case.